server/src/main/java/org/apache/druid/metadata/SQLMetadataConnector.java

```diff
@@ -259,8 +264,15 @@ public void createSegmentTable(final String tableName)
                 + ")",
                 tableName, getPayloadType(), getQuoteString()
             ),
-            StringUtils.format("CREATE INDEX idx_%1$s_datasource ON %1$s(dataSource)", tableName),
-            StringUtils.format("CREATE INDEX idx_%1$s_used ON %1$s(used)", tableName)
+            StringUtils.format(
+                "CREATE INDEX idx_%1$s_datasource_end ON %1$s(dataSource, %2$send%2$s)",
+                tableName,
+                getQuoteString()
+            ),
+            StringUtils.format(
+                "CREATE INDEX idx_%1$s_datasource_sequence ON %1$s(dataSource, sequence_name)",
```

Comment:
@gianm sorry,  I did not review these code carefully. There are no `sequence_name ` of `segmentTable`, `pendingSegmentsTable` does...

Reply:
Ouch, you're right. Thanks for the catch- I will fix it. I'm surprised the tests didn't catch it - I thought they ran these create table commands. I will look into that too.

Reply:
Raised #6356 to fix this. The PR description also explains why the tests didn't catch this.